test/parallel/test-fs-rmdir-recursive.js

```diff
@@ -7,6 +7,9 @@ const fs = require('fs');
 const path = require('path');
 const { validateRmdirOptions } = require('internal/fs/utils');
 
+const realEmitWarning = process.emitWarning;
+process.emitWarning = () => { throw new Error('deprecated'); };
```

Comment:
might be worth checking if we have another test that asserts against an emitted warning, to see if we're using a similar approach.

Reply:
I did copy this from another test. It still only triggers on the first usage of `rmdir` recursive though, which makes testing all the different warning scenarios hard.